### PR TITLE
coverity fixes - util, rxd, sockets, verbs, osd

### PR DESCRIPTION
**Pull request by shefty**
@jithinjosepkl @a-ilango - please confirm that the fixes are correct for your providers

### CODE REVIEW

**Comment by jithinjosepkl** [comment]
Thanks 👍

File: prov/verbs/src/verbs_domain.c [hunk 1]
```diff
@@ -122,16 +122,15 @@ fi_ibv_mr_reg(struct fid *fid, const void *buf, size_t len,
...
+	int ret;
 
 	domain = container_of(fid, struct fi_ibv_domain, domain_fid.fid);
 
 	if (domain->rdm) {
-		errno = 0;
-		rdma_destroy_ep(domain->rdm_cm->listener);
-		if (errno) {
+		ret = rdma_destroy_ep(domain->rdm_cm->listener);
+		if (ret) {
```
Comment (jithinjosepkl) [inline prov/verbs/src/verbs_domain.c]: Travis showing build errors with verbs provider. Looks like rdma_destroy_ep() does not return int.

Comment (a-ilango) [inline prov/verbs/src/verbs_domain.c]: Yes. It's interesting that the man pages available on the internet (linux.die.net for example) say rdma_destroy_ep returns int when it doesn't.

**Comment by shefty** [comment]
I actually pulled up the man pages just for this reason.  Oh well, I'll adjust the patch.  Whoever wrote those man pages should be bonked on the head.

**Comment by shefty** [comment]
D'oh, I added a new commit to this series, that was meant as a separate PR.  The new commit attempts to clarify the domain attribute maximums, such as cq_cnt, ep_cnt, tx/rx_ctx_cnt.

**Pull request merged**